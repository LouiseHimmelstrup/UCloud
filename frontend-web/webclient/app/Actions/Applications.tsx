import { Cloud } from "../../authentication/SDUCloudObject";
import { failureNotification } from "../UtilityFunctions";
import {
    RECEIVE_APPLICATIONS,
    SET_APPLICATIONS_LOADING,
    UPDATE_APPLICATIONS
} from "../Reducers/Applications";

import {
    Application,
    Action,
    SetLoadingAction,
    Page
} from "../Types";
import { emptyPage } from "../DefaultObjects";

interface ReceiveApplicationsAction extends Action { page: Page<Application> }
const receiveApplications = (page: Page<Application>): ReceiveApplicationsAction => ({
    type: RECEIVE_APPLICATIONS,
    page
});

export const fetchApplications = (page: number, itemsPerPage: number): Promise<ReceiveApplicationsAction> =>
    Cloud.get(`/hpc/apps?page=${page}&itemsPerPage=${itemsPerPage}`).then(({ response }: { response: Page<Application> }) => {
        return receiveApplications(response);
    }).catch(() => {
        failureNotification("An error occurred while retrieving applications.")
        return receiveApplications(emptyPage);
    });

/**
 * Sets the loading state for the applications component
 * @param {boolean} loading loading state for the applications component
 */
export const setLoading = (loading: boolean): SetLoadingAction => ({
    type: SET_APPLICATIONS_LOADING,
    loading
});

export const updateApplications = (page: Page<Application>): ReceiveApplicationsAction => ({
    type: UPDATE_APPLICATIONS,
    page
});