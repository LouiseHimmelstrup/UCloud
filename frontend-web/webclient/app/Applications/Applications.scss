.stream {
    height: 500px;
    overflow: auto;
}

.input-group > .form-control.time-selection {
    width: 100px;
}

.text-area-no-resize {
    resize: none;
}

.job-result-box {
    padding-bottom: 16px;
}

.cards.card-margin {
   margin-top: 20px
}