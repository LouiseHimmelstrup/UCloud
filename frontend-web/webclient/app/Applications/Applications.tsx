import * as React from "react";
import { Link } from "react-router-dom";
import * as Pagination from "Pagination";
import { Card, Icon, Rating, List } from "semantic-ui-react";
import { connect } from "react-redux";
import {
    fetchApplications,
    setLoading,
    updateApplications
} from "./Redux/ApplicationsActions";
import { updatePageTitle } from "Navigation/Redux/StatusActions";
import { Page } from "Types";
import { Application } from ".";
import { ApplicationsProps, ApplicationsOperations, ApplicationsStateProps } from ".";
import { setErrorMessage } from "./Redux/ApplicationsActions";
// Requires at least TS 3.0.0
import { MaterialColors } from "Assets/materialcolors.json";
import { favoriteApplicationFromPage } from "Utilities/ApplicationUtilities";
import { Cloud } from "Authentication/SDUCloudObject";
import { setPrioritizedSearch } from "Navigation/Redux/HeaderActions";

const COLORS_KEYS = Object.keys(MaterialColors);

// We need dynamic import due to nature of the import
const blurOverlay = require("Assets/Images/BlurOverlayByDan.png");

class Applications extends React.Component<ApplicationsProps> {

    componentDidMount() {
        const { props } = this;
        props.updatePageTitle();
        props.prioritizeApplicationSearch();
        props.setLoading(true);
        if (this.props.page.items.length === 0) {
            props.fetchApplications(props.page.pageNumber, props.page.itemsPerPage);
        }
    }

    render() {
        const { page, loading, fetchApplications, onErrorDismiss, updateApplications, error } = this.props;
        const favoriteApp = (app: Application) => updateApplications(favoriteApplicationFromPage(app, page, Cloud));
        return (
            <React.StrictMode>
                <Pagination.List
                    loading={loading}
                    onErrorDismiss={onErrorDismiss}
                    errorMessage={error}
                    onRefresh={() => fetchApplications(page.pageNumber, page.itemsPerPage)}
                    pageRenderer={({ items }: Page<Application>) =>
                        <Card.Group className="card-margin">
                            {items.map((app, index) => <SingleApplication key={index} app={app} favoriteApp={favoriteApp} />)}
                        </Card.Group>
                    }
                    page={page}
                    onItemsPerPageChanged={(size) => fetchApplications(0, size)}
                    onPageChanged={(pageNumber) => fetchApplications(pageNumber, page.itemsPerPage)}
                />
            </React.StrictMode >);
    }
}

interface SingleApplicationProps { app: Application, favoriteApp?: (app: Application) => void }
export function SingleApplication({ app, favoriteApp }: SingleApplicationProps) {
    const hashCode = toHashCode(app.description.info.name);
    const color = COLORS_KEYS[(hashCode % COLORS_KEYS.length)];
    const mClength = MaterialColors[color].length;
    const hex = MaterialColors[color][(hashCode % mClength)];
    const even = app.modifiedAt % 2 === 0;
    const opacity = even ? 0.3 : 1;
    const { description } = app.description;
    const image = even ? blurOverlay : `https://placekitten.com/200/200`;
    return (
        <Card>
            {/* FIXME: Move styling to .scss file where possible */}
            <div style={{
                background: hex
            }}>
                <Link to={`/appDetails/${app.description.info.name}/${app.description.info.version}/`}>
                    <div style={{
                        opacity: opacity,
                        width: "100%",
                        height: "200px",
                        backgroundImage: `url('${image}')`,
                        backgroundSize: "cover",
                        backgroundPosition: "center"
                    }} />
                </Link>
            </div>
            <Card.Content>
                <List horizontal floated="right">
                    {!!favoriteApp ? <List.Item>
                        <Rating icon={"star"} maxRating={1} rating={app.favorite ? 1 : 0} onClick={() => !!favoriteApp ? favoriteApp(app) : null} />
                    </List.Item> : null}
                    <List.Item>
                        <Link to={`/applications/${app.description.info.name}/${app.description.info.version}/`}>
                            <Icon color="green" name="play" />
                        </Link>
                    </List.Item>
                </List>
                <Card.Header
                    as={Link}
                    to={`/appDetails/${app.description.info.name}/${app.description.info.version}/`}
                    content={app.description.title}
                />
                <Card.Meta content={app.description.info.version} />
            </Card.Content>
            <Card.Content extra>
                {description.length > 72 ? `${description.slice(0, 72)}...` : description}
            </Card.Content>
        </Card>
    );
}

function toHashCode(name: string): number {
    let hash = 0;
    if (name.length == 0) {
        return hash;
    }
    for (let i = 0; i < name.length; i++) {
        let char = name.charCodeAt(i);
        hash = ((hash << 5) - hash) + char;
        hash = hash & hash; // Convert to 32bit integer
    }
    return Math.abs(hash);
}

const mapDispatchToProps = (dispatch): ApplicationsOperations => ({
    prioritizeApplicationSearch: () => dispatch(setPrioritizedSearch("applications")),
    onErrorDismiss: () => dispatch(setErrorMessage()),
    updatePageTitle: () => dispatch(updatePageTitle("Applications")),
    setLoading: (loading: boolean) => dispatch(setLoading(loading)),
    fetchApplications: (pageNumber: number, itemsPerPage: number) => dispatch(fetchApplications(pageNumber, itemsPerPage)),
    updateApplications: (applications: Page<Application>) => dispatch(updateApplications(applications))
});

const mapStateToProps = ({ applications }): ApplicationsStateProps => ({
    favCount: applications.page.items.filter(it => it.favorite).length,
    ...applications
});

export default connect(mapStateToProps, mapDispatchToProps)(Applications);