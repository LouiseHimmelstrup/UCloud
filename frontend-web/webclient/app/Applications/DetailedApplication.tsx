import * as React from "react";
import PromiseKeeper from "PromiseKeeper";
import { Cloud } from "Authentication/SDUCloudObject";
import { Grid, Header, Table, Label, Icon, List, Rating, Button, Message } from "semantic-ui-react";
import { Link } from "react-router-dom";
import * as ReactMarkdown from "react-markdown";
import { DefaultLoading } from "LoadingIcon/LoadingIcon";
import { ApplicationInformation, ParameterTypes } from "Applications";

type DetailedApplicationProps = any
type DetailedApplicationState = {
    appInformation?: ApplicationInformation
    promises: PromiseKeeper
    loading: boolean
    error?: string
}


class DetailedApplication extends React.Component<DetailedApplicationProps, DetailedApplicationState> {
    constructor(props) {
        super(props);
        this.state = {
            promises: new PromiseKeeper,
            loading: false
        }
    }

    componentDidMount() {
        this.retrieveApplication();
    }

    retrieveApplication() {
        this.setState(() => ({ loading: true }));
        const { appName, appVersion } = this.props.match.params;
        const { promises } = this.state;
        promises.makeCancelable(Cloud.get(`/hpc/apps/${appName}/${appVersion}`))
            .promise.then(({ response }: { response: ApplicationInformation }) =>
                this.setState(() => ({
                    appInformation: response,
                    loading: false,
                }))
            ).catch(err => this.setState({
                error: `An error occurred fetching ${appName}`,
                loading: false
            }));
    }

    favoriteApplication = (): void => {
        const { appInformation } = this.state;
        if (!appInformation) return;
        appInformation.favorite = !appInformation.favorite;
        if (appInformation.favorite) {
            // post
        } else {
            // delete
        }
        this.setState(() => ({ appInformation }));
    }

    render() {
        const { appInformation } = this.state;
        const error = this.state.error ? (
            <Message color="red" content={this.state.error} onDismiss={() => this.setState(() => ({ error: undefined }))} />
        ) : null;
        return (
            <Grid container columns={16}>
                <DefaultLoading loading={this.state.loading} />
                <Grid.Column width={16}>
                    {error}
                    <ApplicationHeader favoriteApplication={this.favoriteApplication} appInformation={appInformation} />
                    <Header as="h3" content="Tags" />
                    <ApplicationTags tags={[] as string[]} />
                    <Header as="h1" content="Tools" />
                    <ApplicationTools appInformation={appInformation} />
                </Grid.Column>
            </Grid>
        );
    }
}


const ApplicationTags = (props: { tags: string[] }) => {
    const mockedTags = ["nanomachines", "medication", "megamachines", "hyper light simulation", "teleportation research"];
    return (
        <React.Fragment>
            {mockedTags.map((tag, i) => <Label key={i} basic content={tag} />)}
        </React.Fragment>
    )
};

interface ApplicationDetails { appInformation?: ApplicationInformation }
const ApplicationTools = ({ appInformation }: ApplicationDetails) => {
    if (appInformation == null) return null;

    const { tool } = appInformation;
    const { hours, minutes, seconds } = tool.description.defaultMaxTime;
    const padNumber = (val: number): string => val < 10 ? `0${val}` : `${val}`;
    const timeString = `${padNumber(hours)}:${padNumber(minutes)}:${padNumber(seconds)}`;
    return (
        <List>
            <List.Item>
                <List.Content floated="left">
                    <Label color="green">
                        <Icon name="wrench" />
                        Container: {tool.description.backend}
                    </Label>
                    <Label color="blue">
                        <Icon name="file" />
                        Output files: {appInformation.description.outputFileGlobs.join(", ")}
                    </Label>
                    <Label content={`${appInformation.description.parameters.length} parameters`} />
                </List.Content>
                <List.Content floated="right">
                    <Label basic>
                        <Icon name="clock" />
                        Default job time: {timeString}
                    </Label>
                    <Label basic>
                        Default number of nodes: {tool.description.defaultNumberOfNodes}
                    </Label>
                    <Label basic content={`Default tasks per node: ${tool.description.defaultTasksPerNode}`} />
                </List.Content>
            </List.Item>
        </List >
    )
}

const typeToString = (parameterType: ParameterTypes): string => {
    switch (parameterType) {
        case ParameterTypes.Integer:
            return "Integer";
        case ParameterTypes.FloatingPoint:
            return "Floating point";
        case ParameterTypes.Text:
            return "Text";
        case ParameterTypes.Boolean:
            return "Boolean";
        case ParameterTypes.InputFile:
            return "Input file";
        case ParameterTypes.InputDirectory:
            return "Input directory";
        default:
            console.warn(`Unhandled parameter type: ${parameterType}`);
            return "";
    }
}

interface ApplicationHeaderProps extends ApplicationDetails { favoriteApplication: () => void }
const ApplicationHeader = ({ appInformation, favoriteApplication }: ApplicationHeaderProps) => {
    if (appInformation == null) return null;
    const { info } = appInformation.description;
    // Not a very good pluralize function.
    const pluralize = (array, text) => (array.length > 1) ? text + "s" : text;
    let authorString = (!!appInformation.description.authors) ? appInformation.description.authors.join(", ") : "";

    return (
        <Header as="h1">
            <Header.Content className="float-right">
                <Button as={Link} basic color="blue" content="Run Application" to={`/applications/${info.name}/${info.version}/`} />
            </Header.Content>
            <Header.Content>
                {appInformation.description.title}
                <span className="app-favorite-padding">
                    <Rating
                        icon="star"
                        size="huge"
                        rating={appInformation.favorite ? 1 : 0}
                        maxRating={1}
                        onClick={() => favoriteApplication()}
                    />
                </span>
                <h4>{appInformation.description.info.version}</h4>
                <h4>{pluralize(appInformation.description.authors, "Author")}: {authorString}</h4>
            </Header.Content>
            <Header.Subheader>
                <ReactMarkdown source={appInformation.description.description} />
            </Header.Subheader>
        </Header>
    );
};

export default DetailedApplication;