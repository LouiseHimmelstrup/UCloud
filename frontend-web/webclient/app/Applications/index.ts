import { File } from "Files";
import { Page } from "Types";
import { match } from "react-router";
import PromiseKeeper from "PromiseKeeper";
import { History } from "history";
import { DetailedResultReduxObject, ApplicationReduxObject } from "DefaultObjects";

export type ApplicationsProps = ApplicationReduxObject & ApplicationsOperations;

export interface Analysis {
    status: string
    state: string
    jobId: string
    appName: string
    appVersion: string
    createdAt: number
    modifiedAt: number
    owner: string
}

export interface ApplicationsOperations {
    prioritizeApplicationSearch: () => void
    onErrorDismiss: () => void
    updatePageTitle: () => void
    setLoading: (loading: boolean) => void
    setFavoritesLoading: (loading: boolean) => void
    fetchApplications: (a: number, b: number) => void
    fetchFavorites: (a: number, b: number) => void
    receiveApplications: (applications: Page<Application>) => void
}

export interface AnalysesProps extends AnalysesStateProps, AnalysesOperations { }

export interface AnalysesStateProps {
    page: Page<Analysis>
    loading: boolean
    error?: string
}

export interface AnalysesOperations {
    onErrorDismiss: () => void
    updatePageTitle: (title: string) => void
    setLoading: (loading: boolean) => void
    fetchAnalyses: (itemsPerPage: number, pageNumber: number) => void
}

export interface AnalysesState {
    reloadIntervalId: number
}

export interface DetailedResultOperations {
    receivePage: (page: Page<File>) => void,
    setPageTitle: (jobId: string) => void
    setLoading: (loading: boolean) => void
    detailedResultError: (error: string) => void
    fetchPage: (jobId: string, pageNumber: number, itemsPerPage: number) => void
}

export interface DetailedResultProps extends DetailedResultReduxObject, DetailedResultOperations {
    match: match<{ jobId: string }>
    history: History
}

export interface Application {
    favorite: boolean
    owner: string
    createdAt: number
    modifiedAt: number
    description: ApplicationDescription
    tool: ApplicationTool
}

interface ApplicationTool {
    owner: string
    createdAt: number
    modifiedAt: number
    description: {
        info: ApplicationInfo
        container: string
        defaultNumberOfNodes: number
        defaultTasksPerNode: number
        defaultMaxTime: MaxTime
        requiredModules: string[]
        authors: string[]
        title: string
        description: string
        backend: string
    }
}

interface ApplicationInfo {
    name: string
    version: string
}

export interface ApplicationDescription {
    info: ApplicationInfo
    tool: ApplicationInfo
    authors: string[]
    title: string
    description: string
    invocation: any[]
    parameters: ApplicationParameter[]
    outputFileGlobs: string[]
}

export interface DetailedResultState {
    complete: boolean
    appState: string
    status: string
    app: {
        name: string
        version: string
    }
    stdout: string
    stderr: string
    stdoutLine: number
    stderrLine: number
    stdoutOldTop: number,
    stderrOldTop: number,
    reloadIntervalId: number
    promises: PromiseKeeper
}

export type StdElement = { scrollTop: number, scrollHeight: number } | null

export type MaxTime = {
    hours: number | null
    minutes: number | null
    seconds: number | null
} | null

export interface JobInfo {
    maxTime: MaxTime
    numberOfNodes: number | null
    tasksPerNode: number | null
}

export interface RunAppState {
    promises: PromiseKeeper
    error?: string
    loading: boolean
    appName: string
    favorite: boolean
    displayAppName: string
    appVersion: string
    appDescription: string
    appAuthor: string[]
    parameters: ApplicationParameter[]
    parameterValues: {}
    jobInfo: JobInfo
    tool: {} // ???
    comment: string
    jobSubmitted: boolean
}

export interface RunAppProps {
    match: match<{ appName: string, appVersion: string }>
    history: History
    updatePageTitle: () => void
}

export interface NumberParameter extends BaseParameter {
    defaultValue: { value: number, type: "double" | "int" } | null
    min: number | null
    max: number | null
    step: number | null
    type: "integer" | "floating_point"
}

export interface BooleanParameter extends BaseParameter {
    defaultValue: { value: boolean, type: "bool" } | null
    trueValue?: string | null
    falseValue?: string | null
    type: "boolean"
}

export interface InputFileParameter extends BaseParameter {
    defaultValue: string | null
    type: "input_file"
}

export interface InputDirectoryParameter extends BaseParameter {
    defaultValue: string | null
    type: "input_directory"
}

export interface TextParameter extends BaseParameter {
    defaultValue: { value: string, type: "string" } | null
    type: "text"
}

interface BaseParameter {
    name: string
    optional: boolean
    title: string
    description: string
    unitName?: string | null
}

export type ApplicationParameter = InputFileParameter | InputDirectoryParameter | NumberParameter | BooleanParameter | TextParameter

type Invocation = WordInvocation | VarInvocation

interface WordInvocation {
    type: "word"
    word: string
}

interface VarInvocation {
    type: "var"
    variableNames: string[]
    prefixGlobal: string
    suffixGlobal: string
    prefixVariable: string
    suffixVariable: string
    variableSeparator: string
}

type Info = { name: string, version: string }
export interface Description {
    info: Info
    tool: Info
    authors: string[]
    title: string
    description: string
    invocation: Invocation[]
    parameters: ApplicationParameter[]
    outputFileGlobs: [string, string]
}
interface Tool {
    owner: string
    createdAt: number
    modifiedAt: number
    description: ToolDescription
}

interface ToolDescription {
    info: Info
    container: string
    defaultNumberOfNodes: number,
    defaultTasksPerNode: number,
    defaultMaxTime: {
        hours: number
        minutes: number
        seconds: number
    }
    requiredModules: any[],
    authors: string[]
    title: string,
    description: string
    backend: string
}


export interface ApplicationInformation {
    owner: string
    favorite?: boolean
    createdAt, modifiedAt: number
    description: Description
    tool: Tool
}

export enum ParameterTypes {
    InputFile = "input_file",
    InputDirectory = "input_directory",
    Integer = "integer",
    FloatingPoint = "floating_point",
    Text = "text",
    Boolean = "boolean"
}

export interface SearchFieldProps {
    onSubmit: () => void
    icon: string
    placeholder: string
    value: string
    loading: boolean
    onValueChange: (value: string) => void
}