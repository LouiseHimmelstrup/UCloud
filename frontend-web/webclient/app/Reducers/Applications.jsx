export const RECEIVE_APPLICATIONS = "RECEIVE_APPLICATIONS";
export const SET_LOADING = "SET_LOADING";
export const TO_PAGE = "TO_PAGE";
export const UPDATE_APPLICATIONS_PER_PAGE = "UPDATE_APPLICATIONS_PER_PAGE";
export const UPDATE_APPLICATIONS = "UPDATE_APPLICATIONS";

const applications = (state = [], action) => {
    switch (action.type) {
        case RECEIVE_APPLICATIONS: {
            return { ...state, applications: action.applications, loading: false };
        }
        case SET_LOADING: {
            return { ...state, loading: action.loading };
        }
        case TO_PAGE: {
            return { ...state, currentApplicationsPage: action.pageNumber };
        }
        case UPDATE_APPLICATIONS_PER_PAGE: {
            return { ...state, applicationsPerPage: action.applicationsPerPage };
        }
        case UPDATE_APPLICATIONS: {
            return { ...state, applications: action.applications };
        }
        default: {
            return state;
        }
    }
};

export default applications;