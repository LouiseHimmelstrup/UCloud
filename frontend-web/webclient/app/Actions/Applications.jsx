import { Cloud } from "../../authentication/SDUCloudObject";
import { RECEIVE_APPLICATIONS, SET_LOADING, UPDATE_APPLICATIONS_PER_PAGE } from "../Reducers/Applications";

const receiveApplications = (applications) => ({
    type: RECEIVE_APPLICATIONS,
    applications
});

export const fetchApplications = () =>
    Cloud.get("/hpc/apps").then(({ response }) => {
        response.sort((a, b) =>
            a.prettyName.localeCompare(b.prettyName)
        );
        return receiveApplications(response);
    });

export const setLoading = (loading) => ({
    type: SET_LOADING,
    loading
});

export const toPage = (pageNumber) => ({
    type: TO_PAGE,
    pageNumber
});

export const updateApplications = (applications) => ({
    type: UPDATE_APPLICATIONS,
    applications
});

export const updateApplicationsPerPage = (applicationsPerPage) => ({
    type: UPDATE_APPLICATIONS_PER_PAGE,
    applicationsPerPage
});