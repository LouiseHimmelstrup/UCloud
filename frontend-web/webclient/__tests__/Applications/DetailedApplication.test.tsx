import * as React from "react";
import DetailedApplication from "Applications/DetailedApplication";
import { create } from "react-test-renderer";
import { mount, shallow } from "enzyme";
import { Message } from "semantic-ui-react";
import { configure } from "enzyme";
import { detailedApplication } from "../mock/Applications";
import * as Adapter from "enzyme-adapter-react-16";
import { MemoryRouter } from "react-router";

configure({ adapter: new Adapter() });

describe("Detailed application", () => {
    test("Mount component", () => {
        expect(create(
            <DetailedApplication match={{ params: { appName: "someName", appVersion: "someVersion" } }} />
        ).toJSON()).toMatchSnapshot();
    });

    test("Set error message", () => {
        const error = "Error Message!";
        const detailedApp = mount(<DetailedApplication match={{ params: { appName: "someName", appVersion: "someVersion" } }} />);
        detailedApp.setState(() => ({ error }));
        expect(detailedApp.state("error")).toBe(error);
        expect(detailedApp.find(Message).props().content).toBe(error);
        detailedApp.find(Message).find("i").simulate("click");
        expect(detailedApp.find(Message).exists()).toBe(false);
    });

    test("Component with application", () => {
        let detailedAppWrapper = mount(<MemoryRouter><DetailedApplication match={{ params: { appName: "someName", appVersion: "someVersion" } }} /></MemoryRouter>);
        detailedAppWrapper = detailedAppWrapper.update();
        detailedAppWrapper.find(DetailedApplication).instance().setState({ appInformation: detailedApplication, loading: false });
        expect(detailedAppWrapper.html()).toMatchSnapshot();
    });
});