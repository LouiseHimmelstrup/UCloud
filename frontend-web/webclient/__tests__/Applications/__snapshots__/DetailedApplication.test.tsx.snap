// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detailed application Component with application 1`] = `"<div class=\\"sc-ifAKCX iCcktw\\"><div width=\\"1,0.8125\\" class=\\"sc-bdVaJa iAxnNK\\"><div class=\\"sc-eNQAEJ dPxNMT\\"><a href=\\"/applications/figlet-count/1.0.0/\\"><button color=\\"blue\\" class=\\"sc-chPdSV koOMMT\\">Run Application</button></a></div><h1 font-size=\\"6\\" class=\\"sc-VigVT dzBifV\\">Figlet Counter<span class=\\"app-favorite-padding\\"><svg style=\\"vertical-align: center;\\" cursor=\\"auto\\" class=\\"sc-gZMcBi cRLIbN sc-iwsKbI dwZotm\\" viewBox=\\"0 0 23 22\\" width=\\"24\\" height=\\"24\\" fill=\\"currentcolor\\"><path d=\\"M14.939 7.146l7.886 1.146-5.707 5.562 1.347 7.854L11.412 18 4.36 21.708l1.347-7.854L0 8.292l7.886-1.146L11.412 0l3.527 7.146zM9.878 9.888l-3.432.498 2.483 2.42-.586 3.418 3.07-1.614 3.068 1.614-.586-3.417 2.483-2.42-3.431-.499-1.535-3.11-1.534 3.11z\\" fill=\\"red\\"></path></svg></span><h4>1.0.0</h4><h4>Author: Dan Sebastian Thrane &lt;dthrane@imada.sdu.dk&gt;</h4><h5 font-size=\\"2\\" class=\\"sc-cSHVUG jOGwme\\"><p>Count with Figlet!</p></h5></h1><h3 font-size=\\"4\\" class=\\"sc-fjdhpX gjFQeq\\">Tags</h3><div color=\\"black\\" font-size=\\"0\\" class=\\"sc-csuQGl jbfBdG\\">nanomachines</div><div color=\\"black\\" font-size=\\"0\\" class=\\"sc-csuQGl jbfBdG\\">medication</div><div color=\\"black\\" font-size=\\"0\\" class=\\"sc-csuQGl jbfBdG\\">megamachines</div><div color=\\"black\\" font-size=\\"0\\" class=\\"sc-csuQGl jbfBdG\\">hyper light simulation</div><div color=\\"black\\" font-size=\\"0\\" class=\\"sc-csuQGl jbfBdG\\">teleportation research</div><h3 font-size=\\"4\\" class=\\"sc-fjdhpX gjFQeq\\">Tools</h3><div class=\\"sc-ifAKCX hEFUra\\"><div color=\\"white\\" font-size=\\"0\\" class=\\"sc-csuQGl cIWzZI\\"><div class=\\"sc-bdVaJa guxnrH\\"><i class=\\"fas fa-wrench\\"></i></div>Container: SINGULARITY</div><div color=\\"white\\" font-size=\\"0\\" class=\\"sc-csuQGl hcRoGT\\"><div class=\\"sc-bdVaJa guxnrH\\"><i class=\\"far fa-file\\"></i></div>Output files: stdout.txt, stderr.txt</div><div color=\\"black\\" font-size=\\"0\\" class=\\"sc-csuQGl jbfBdG\\">1 parameters</div><div class=\\"sc-bdVaJa gtvdIr\\"></div><div color=\\"black\\" font-size=\\"0\\" class=\\"sc-csuQGl jbfBdG\\"><i class=\\"far fa-clock\\"></i>Default job time: 00:01:00</div><div color=\\"black\\" font-size=\\"0\\" class=\\"sc-csuQGl jbfBdG\\">Default number of nodes: 1</div><div color=\\"black\\" font-size=\\"0\\" class=\\"sc-csuQGl jbfBdG\\">Default tasks per node: 1</div></div></div><div width=\\"0,0.1875\\" class=\\"sc-eNQAEJ hUvKIa\\"><a href=\\"/applications/figlet-count/1.0.0/\\"><button color=\\"blue\\" class=\\"sc-chPdSV koOMMT\\">Run Application</button></a></div></div>"`;

exports[`Detailed application Mount component 1`] = `
<div
  className="sc-ifAKCX iCcktw"
>
  <div
    className="sc-bdVaJa iAxnNK"
    width={
      Array [
        1,
        0.8125,
      ]
    }
  >
    <div
      className="sc-fjdhpX eIwIdU"
    />
    <div
      className="sc-ifAKCX LORpy"
    >
      <div
        className="sc-bdVaJa jqyTwv"
        width={0.7}
      >
        <i
          className="fas fa-circle-notch fa-spin"
        />
      </div>
    </div>
  </div>
  <div
    className="sc-fjdhpX jrxGCm"
    width={
      Array [
        0,
        0.1875,
      ]
    }
  >
    <div
      className="sc-bdVaJa gvJQvB"
    />
  </div>
</div>
`;
